### PR TITLE
deploy current develop to aws to allow faster testing 

### PR DESCRIPTION
**Pull request by linglp**
PR Checklist:
[ ] Describe and explain your intentions for this change
[ ] Setup pre-commit and run the validators (info in README.md)
Per slack comment: https://sagebionetworks.slack.com/archives/C07HLTBS5D2/p1729016982033769

Also fixed workflow related to: https://sagebionetworks.jira.com/browse/FDS-2514
The initial error message in the workflow: 
```
Run pre-commit/action@v3.0.0
Run python -m pip install pre-commit
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.[12](https://github.com/Sage-Bionetworks/schematic-infra/actions/runs/11353454017/job/31578496293#step:4:14)/README.venv for more information.
```

### CODE REVIEW

File: .github/workflows/aws-deploy.yml [hunk 1]
```diff
@@ -18,8 +18,20 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
-      - name: Static Analysis
-        uses: pre-commit/action@v3.0.0
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
```
Comment (zaro0508) [inline .github/workflows/aws-deploy.yml]: if you want to keep everything as-is i suggest you simply add a `setup-python` action.. 

```
      - uses: actions/setup-python@v5
        with:
          python-version: "3.x"
```

**Review by zaro0508 — COMMENTED** [review]
I believe the root cause of the pre-commit problem and the fix is here: https://github.com/pre-commit/action/issues/210#issuecomment-2378269371

Your change will work however it seems much more verbose than the original setup.

File: .github/workflows/aws-deploy.yml [hunk 2]
```diff
@@ -28,11 +40,8 @@ jobs:
           actions_comment: false
           cdk_args: '--context env=dev --debug'
 
-      - name: Install dependencies
-        run: pip install -r requirements.txt -r requirements-dev.txt
-
       - name: Run unit tests
-        run: python -m pytest tests/ -s -v
+        run: venv/bin/python -m pytest tests/ -s -v
```
Comment (BryanFauble) [inline .github/workflows/aws-deploy.yml]: This is not really "unit" tests anymore. It's everything (Unit and integration tests).

Also - Do we get value out of these running? Some thoughts:

1. In order to build the docker container in the first place **we would have needed to get a successful build**. Perhaps we should really **only run unit tests here as a cursory check** to verify that dependencies are working correctly, but not that the integration tests are still functioning.
2. With the changes we're working on in https://github.com/Sage-Bionetworks/schematic/pull/1512 - The intention is that we will be able to run a number of integration tests against the dev version of schematic deployed to AWS. Perhaps this is enough for us to consider only running the unit test directory here `tests/unit/`

Comment (BryanFauble) [reply .github/workflows/aws-deploy.yml]: I glossed over the fact that it's this directory: https://github.com/Sage-Bionetworks/schematic-infra/tree/dev/tests that is being run - NOT, the testing directory of the schematic API/python code

**Review by BryanFauble — APPROVED** [review]
LGTM!

File: .github/workflows/aws-deploy.yml (hunk 1 above)
Comment (linglp) [reply .github/workflows/aws-deploy.yml]: Thanks. The solution that you proposed worked. I will keep everything as is then. Thank you Khai!

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**